x/stakingplus/keeper/keeper_test.go

```diff
@@ -0,0 +1,133 @@
+	"github.com/Finschia/finschia-sdk/x/stakingplus/testutil"
+)
+
+type KeeperTestSuite struct {
+	suite.Suite
+	ctx sdk.Context
+
+	app           *runtime.App
+	accountKeeper authkeeper.AccountKeeper
+	bankKeeper    bankkeeper.Keeper
+	stakingKeeper *stakingkeeper.Keeper
+	msgServer     stakingtypes.MsgServer
+
+	stranger sdk.AccAddress
+	grantee  sdk.AccAddress
+
+	balance math.Int
+}
+
+func (s *KeeperTestSuite) SetupTest() {
+	ctrl := gomock.NewController(s.T())
+	foundationKeeper := testutil.NewMockFoundationKeeper(ctrl)
+
+	app, err := simtestutil.Setup(
+		depinject.Configs(
+			testutil.AppConfig,
+			depinject.Supply(log.NewNopLogger()),
+		),
+		&s.accountKeeper,
+		&s.bankKeeper,
+		&s.stakingKeeper,
+	)
+	s.Require().NoError(err)
+
+	s.app = app
+	s.ctx = s.app.BaseApp.NewContext(false)
+
+	s.msgServer = keeper.NewMsgServerImpl(s.stakingKeeper, foundationKeeper)
+
+	createAddress := func() sdk.AccAddress {
+		return sdk.AccAddress(secp256k1.GenPrivKey().PubKey().Address())
+	}
+
+	s.stranger = createAddress()
+	s.grantee = createAddress()
+
+	s.balance = math.NewInt(1000000)
+	holders := []sdk.AccAddress{
+		s.stranger,
+		s.grantee,
+	}
+	for _, holder := range holders {
+		amount := sdk.NewCoins(sdk.NewCoin(sdk.DefaultBondDenom, s.balance))
+
+		// using minttypes here introduces dependency on x/mint
+		// the work around would be registering a new module account on this suite
+		// because x/bank already has dependency on x/mint, and we must have dependency
+		// on x/bank, it's OK to use x/mint here.
+		minterName := minttypes.ModuleName
+		err := s.bankKeeper.MintCoins(s.ctx, minterName, amount)
+		s.Require().NoError(err)
+
+		minter := s.accountKeeper.GetModuleAccount(s.ctx, minterName).GetAddress()
+		err = s.bankKeeper.SendCoins(s.ctx, minter, holder, amount)
+		s.Require().NoError(err)
+	}
+
+	// approve Msg/CreateValidator to grantee
+	foundationKeeper.
+		EXPECT().
+		Accept(gomock.Any(), s.grantee, NewCreateValidatorAuthorizationMatcher(s.grantee)).
+		Return(nil)
+	foundationKeeper.
+		EXPECT().
+		Accept(gomock.Any(), gomock.Any(), gomock.Any()).
+		Return(sdkerrors.ErrUnauthorized)
+}
+
+func TestKeeperTestSuite(t *testing.T) {
+	suite.Run(t, new(KeeperTestSuite))
+}
+
+type CreateValidatorAuthorizationMatcher struct {
+	authz stakingplus.CreateValidatorAuthorization
+}
+
+func NewCreateValidatorAuthorizationMatcher(grantee sdk.AccAddress) *CreateValidatorAuthorizationMatcher {
+	return &CreateValidatorAuthorizationMatcher{
+		authz: stakingplus.CreateValidatorAuthorization{
+			ValidatorAddress: sdk.ValAddress(grantee).String(),
+		},
+	}
+}
+
+func (c CreateValidatorAuthorizationMatcher) Matches(x interface{}) bool {
+	msg, ok := x.(sdk.Msg)
+	if !ok {
+		return false
+	}
+
+	resp, err := c.authz.Accept(sdk.Context{}, msg)
+	return resp.Accept && (err == nil)
+}
+
+func (c CreateValidatorAuthorizationMatcher) String() string {
+	return fmt.Sprintf("grants %s to %s", c.authz.MsgTypeURL(), c.authz.ValidatorAddress)
+}
```

Comment:
We don't need this part for mock testing based on previous comment. We can assume our authorization will be registered by foundation module properly.